src/memory_dump/dump.c

```diff
@@ -34,311 +34,23 @@
-    state.last_trace = (state.last_trace + 1) % TRACE_SIZE;
-
-#ifdef _DEBUG
-    debug_addr("MRU Reprotecting page ",
-               (off64_t) state.pages_cache[state.last_page]);
-#endif
-    int result =
-        syscall(SYS_mprotect, (void *)state.pages_cache[state.last_page],
-                PAGESIZE, PROT_NONE);
-    if (result == -1) {
-      //assert(false);
-    }
-  }
-
-  state.pages_cache[state.last_page] = start_of_page;
-  state.last_page = (state.last_page + 1) % state.log_size;
-
-#ifdef _DEBUG
-  debug("     >>>> page cache\n");
-  for (int i = 0; i < state.log_size; i++) {
-    int c = (i + state.last_page) % state.log_size;
-    debug_addr("     >>>> ", (off64_t) state.pages_cache[c]);
-  }
-#endif
-  errno = esaved;
-}
-
-static void fill_dump_name(char *buf, int stack_pos, off64_t addr) {
-  size_t i, j;
-  for (i = 0; state.dump_path[stack_pos][i] != '\0'; i++)
-    buf[i] = state.dump_path[stack_pos][i];
-  buf[i++] = '/';
-
-  buf = append_addr(buf + i, addr);
-
-  const char suffix[] = ".memdump";
-  i = 0;
-  for (j = 0; suffix[j] != '\0'; j++)
-    buf[i++] = suffix[j];
-
-  buf[i++] = '\0';
-}
-
-/* dump_region: dumps to a file "<start>.memdump"
- * the memory segment starting at address <start>
- * and ending at address <end>
- */
-static bool write_page(char path[], off64_t start) {
-  int out = syscall(SYS_openat, AT_FDCWD, path,
-                    O_WRONLY | O_CREAT | O_EXCL, S_IRWXU);
-
-  if (out <= 0) {
-    /* region already dumped */
-    if (errno == EEXIST)
-      return false;
-    else {
-      debug("could not open:");
-      debug(path);
-      debug("\n");
-      exit(EXIT_FAILURE);
-    }
-  }
-
-  int wr = syscall(SYS_write, out, (char *)start, PAGESIZE);
-  if (wr < 0) {
-    syscall(SYS_close, out);
-    int result = syscall(SYS_unlinkat, AT_FDCWD, path, 0);
-    assert(result != -1);
-    return false;
-  }
-  assert(wr == PAGESIZE);
+#include "debug.h"
 
-  syscall(SYS_close, out);
-  return true;
-}
-
-static bool dump_page(off64_t start) {
-  /* Dump Page */
-  char current_path[MAX_PATH];
-  fill_dump_name(current_path, state.stack_pos, start);
-  bool result = write_page(current_path, start);
-  if (result) {
-    /* Link in parent's dumps */
-    for (int i = 0; i < state.stack_pos; i++) {
-      char parent_path[MAX_PATH];
-      fill_dump_name(parent_path, i, start);
-      syscall(SYS_linkat, AT_FDCWD, current_path, AT_FDCWD, parent_path, AT_SYMLINK_FOLLOW);
-    }
-  }
-  return result;
-}
-
-static void dump_handler(int sig, siginfo_t *si, void *unused) {
-
-  int esaved = errno;
-  bool old_status = state.mtrace_active;
-  state.mtrace_active = false;
-  char *touched_addr = si->si_addr;
-  char *start_of_page = round_to_page(touched_addr);
-
-#ifdef _DEBUG
-  debug_addr("DUMP Detected access at: ", (off64_t) touched_addr);
-#endif
-
-  mru_handler(sig, si, unused);
-
-  /* Dump page */
-  dump_page((off64_t) start_of_page);
+static int times_called = 0;
+char writing_buff[1024];
```

Comment:
1024 should use a `#define` constant ? What happens if the name of the region is > 1024 is the error detected, or is this impossible ?